nncf/common/graph/graph.py

```diff
@@ -542,24 +552,24 @@ def _get_graph_for_visualization(self) -> nx.DiGraph:
 
     def get_node_by_name(self, name: NNCFNodeName) -> NNCFNode:
         matches = []
-        for nx_node in self._nx_graph.nodes.values():
-            if nx_node[NNCFGraph.NODE_NAME_ATTR] == name:
-                matches.append(nx_node)
+        for node in self.get_all_nodes():
+            if node.node_name == name:
+                matches.append(node)
         if not matches:
             raise RuntimeError('Could not find a node {} in NNCFGraph!'.format(name))
         if len(matches) > 1:
             raise RuntimeError('More than one node in NNCFGraph matches name {}:\n{}'.
                                format(name,
                                       '\t\n'.join(
-                                          [n[NNCFGraph.KEY_NODE_ATTR] for n in matches])))
-        return self._nx_node_to_nncf_node(next(iter(matches)))
+                                          [str(n.node_id) for n in matches])))
+        return next(iter(matches))
```

Comment:
Why you don't use `matches[0]`?

Reply:
`next(iter())` is a common idiom and also works when the indices are non-integer.